onebusaway-android/src/main/java/org/onebusaway/android/ui/TripPlanFragment.java

```diff
@@ -0,0 +1,464 @@
+                dialog.dismiss();
+
+                boolean optimizeTransfers = minimizeTransfersCheckbox.isChecked();
+
+                int modeId = transitModeResource.getResourceId(spinnerTravelBy.getSelectedItemPosition(), 0);
+
+                boolean wheelchair = wheelchairCheckbox.isChecked();
+
+                String maxWalkString = maxWalkEditText.getText().toString();
+                double maxWalkDistance;
+                if (TextUtils.isEmpty(maxWalkString)) {
+                    maxWalkDistance = Double.MAX_VALUE;
+                } else {
+                    double d = Double.parseDouble(maxWalkString);
+                    maxWalkDistance = unitsAreImperial ? d / 3.281 : d;
+                }
+
+                builder.setOptimizeTransfers(optimizeTransfers)
+                        .setModeSetById(modeId)
+                        .setWheelchairAccessible(wheelchair)
+                        .setMaxWalkDistance(maxWalkDistance);
+            }
+        });
+    }
+
+    private void makeNoLocationToast() {
+        Toast.makeText(getContext(), getString(R.string.tripplanner_error_no_location), Toast.LENGTH_SHORT).show();
+    }
+
+    private CustomAddress makeAddressFromLocation() {
+        Locale locale = Locale.getDefault();
+
+        Location loc = Application.getLastKnownLocation(getContext(), mGoogleApiClient);
+        if (loc == null) {
+            if (getContext() != null)
+                Toast.makeText(getContext(), getString(R.string.main_location_unavailable), Toast.LENGTH_SHORT).show();
+            return null;
+        }
+        CustomAddress address = new CustomAddress(locale);
+        address.setAddressLine(0, getString(R.string.tripplanner_current_location));
+        address.setLatitude(loc.getLatitude());
+        address.setLongitude(loc.getLongitude());
+        return address;
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent intent) {
+        if (resultCode != -1) {
+            Log.e(TAG, "Error getting geocoding results");
+            return;
+        }
+
+        Place place = PlaceAutocomplete.getPlace(getActivity(), intent);
+
+        // note that onResume will run after this function. We need to put new objects in the bundle.
+
+        if (requestCode == FROM_ADDRESS_GEOCODE) {
+            fromAddress = new CustomAddress();
+            builder.setFrom(fromAddress);
+            handlePlaceResult(place, fromAddress, fromAddressTextArea);
+        } else if (requestCode == TO_ADDRESS_GEOCODE) {
+            toAddress = new CustomAddress();
+            builder.setTo(toAddress);
+            handlePlaceResult(place, toAddress, toAddressTextArea);
+        }
+
+    }
+
+    private void handlePlaceResult(Place place, CustomAddress address, TextView tv) {
+        address.setAddressLine(0, place.getName().toString());
+        address.setAddressLine(1, place.getAddress().toString());
+
+        LatLng loc = place.getLatLng();
```

Comment:
`Location` should be used here, again to keep all the Google imports specific to Maps API v2 in the `org.onebusaway.android.map.googlemapsv2` package.  You'll probably need to abstract the implementation of the Places widget into `ProprietaryMapHelpV2` there, and use the Android `Geocoder` for the Amazon build variant in the `ProprietaryMapHelpV2` under `src/amazon/java/org/onebusaway/android/map/googlemapsv2`.